File: 2024_09~12/Data Structure/Mid Exam/queue/linear/queue.c
#include "linear_queue.h"

void print_queue(QUEUE* q)
{
    for(int i = 0; i < MAX/2; i++)
    {
        if(i <= q->front || i > q->rear)
            printf("%3c|", ' ');
        else
            printf("%3d|",q->data[i]);

        if((i + 1) % 10 == 0)
            printf("\n");
    }
    printf("\n");

}

int main(void)
{
    QUEUE q;
    init_queue(&q);

    for(int i = 0; i < MAX/2; i++)
    {
        enqueue(&q, i);  
        
        print_queue(&q);
    }
    for(int i = 0; i < MAX/2; i++)
    {
        dequeue(&q);
        print_queue(&q);
    }

    return 0;
}

File: JAVA/Data_Structure/queue/queue.java
class queue{
	private final int size = 10;
	private int front = 0, rear = 0;
	private boolean count = false;
	private int[] q = new int[size];

	void enqueue(int value){
		if(rear == size - 1)
		{
			rear = 0;
			count = true;
		}

		if((front == rear) && count)
			return;

		q[rear++] = value;
	}

	int dequeue(){
		if(front == size - 1){
			front = 0;
			count = false;
		}
	}
}
